modules/libcom/src/error/errlog.c

```diff
@@ -466,17 +467,38 @@ int eltc(int yesno)
 {
     errlogInit(0);
     epicsMutexMustLock(pvt.msgQueueLock);
-    pvt.toConsole = yesno;
+    pvt.toConsole = yesno ? 1 : 0;
     epicsMutexUnlock(pvt.msgQueueLock);
     errlogFlush();
     return 0;
 }
 
 int errlogSetConsole(FILE *stream)
 {
+    int closeConsole = 0;
     errlogInit(0);
+    if (!stream) {
+        stream = stderr;
+    }
+#ifdef vxWorks
+    /* The stderr/stdout we get may become invalid as they are thread specific.
+     * Also, the FDs are actually maps to some "real" FDs.
+     * Thus, we clone the stream and link it with a dup of the "real" FD.
+     * The dup'ed FD will be closed when the FILE handle is fclosed. */
+    if (fileno(stream) <= 2) {
+        closeConsole = 1;
+        stream = fdopen(dup(ioTaskStdGet(0, fileno(stream))),"w");
```

Comment:
I confirmed that `fileno()` of an invalid stream returns `-1`, that `ioTaskStdGet()` of an invalid FD (not 0,1 or 2) returns `-1`, that `dup(-1)` return `-1` and that `fdopen(-1,"w")` returns `NULL`. 
Also `fclose(NULL)` simply returns `-1` as well as `fprintf(NULL,"x")`.

Reply:
... however the functions taking a `FILE*` argument do not explicitly check for `NULL` arguments but instead access the `unsigned long` at that address and check it against the pointer itself, `0` in this case (on most, maybe all supported archs, depending how `NULL` gets encoded), and then the byte at offset `0x0a` against an enum with value`102`. Not the safest thing in the world....